src/hasi/hasi/meshtomeshregistrar.py

```diff
@@ -72,6 +74,58 @@ def mesh_to_image(self, mesh:MeshType, reference_image:ImageType=None) -> ImageT
         image = distance.GetOutput()
         return image
 
+    # Randomly sample mesh to get a reduced point set
+    # This may not be suitable for low-density meshes
+    def randomly_sample_mesh_points(self, mesh:MeshType, sampling_rate:float=1.0) -> PointSetType:
+        point_set = self.PointSetType.New(Points=mesh.GetPoints())
+
+        if(sampling_rate > 1.0):
+            raise ValueError('Cannot resample with a rate greater than 1.0!')
+        
+        selected_points = set()
+        num_source_points = point_set.GetNumberOfPoints()
+        resampled_set = self.PointSetType.New()
 
+        while resampled_set.GetNumberOfPoints() < sampling_rate * num_source_points:
+            point_index = random.randrange(0, num_source_points)
+            while(point_index in selected_points):
+                point_index = random.randrange(0, num_source_points)
+            selected_points.add(point_index)
 
+            resampled_set.SetPoint(
+                resampled_set.GetNumberOfPoints(),
+                point_set.GetPoint(point_index))
 
+        return resampled_set
+    
+    # TODO minimize memory costs by using itkPolyData
+    @staticmethod
+    def resample_template_from_target(template_mesh:MeshType, target_mesh:MeshType) -> MeshType:
+        # Convert to numpy arrays
+
+        # Set points directly
+        target_pts = vtk.vtkPoints()
+        target_pts.SetNumberOfPoints(target_mesh.GetNumberOfPoints())
+        for i in range(0, target_pts.GetNumberOfPoints()):
+            target_pts.SetPoint(i, target_mesh.GetPoint(i))
+        target_pd = vtk.vtkPolyData()
+        target_pd.SetPoints(target_pts)
+
+        # Do stuff in VTK
+        loc = vtk.vtkKdTreePointLocator()
```

Comment:
Can we use [this](https://itk.org/Doxygen/html/classitk_1_1Statistics_1_1KdTree.html) instead?